app/database/permission_granted_store_compute_all_access.go

```diff
@@ -0,0 +1,146 @@
+					items_items.descendants_and_solution_view_propagation_value+2)) AS can_view_value,
+				MAX(IF(items_items.grant_view_propagation, LEAST(parent.can_grant_view_generated_value, 4), 1)) AS can_grant_view_value,
+				MAX(IF(items_items.watch_propagation, LEAST(parent.can_watch_generated_value, 3), 1)) AS can_watch_value,
+				MAX(IF(items_items.edit_propagation, LEAST(parent.can_edit_generated_value, 3), 1)) AS can_edit_value
+			FROM items_items
+			JOIN permissions_generated AS parent
+				ON parent.item_id = items_items.parent_item_id 
+			JOIN items AS parent_item
+				ON parent_item.id = items_items.parent_item_id
+			WHERE
+				parent.group_id = permissions_propagate.group_id AND items_items.child_item_id = permissions_propagate.item_id
+			GROUP BY permissions_propagate.group_id, permissions_propagate.item_id
+		) AS new_data ON 1
+		LEFT JOIN LATERAL (
+			SELECT
+				MAX(can_view_value) AS can_view_value,
+				MAX(can_grant_view_value) AS can_grant_view_value,
+				MAX(can_watch_value) AS can_watch_value,
+				MAX(can_edit_value) AS can_edit_value,
+				MAX(is_owner) AS is_owner
+			FROM permissions_granted
+			WHERE permissions_granted.group_id = permissions_propagate.group_id AND
+				permissions_granted.item_id = permissions_propagate.item_id
+			GROUP BY permissions_granted.group_id, permissions_granted.item_id
+		) AS granted ON 1
+		WHERE propagate_access = 'self'
+		ON DUPLICATE KEY UPDATE
+			can_view_generated = VALUES(can_view_generated),
+			can_grant_view_generated = VALUES(can_grant_view_generated),
+			can_watch_generated = VALUES(can_watch_generated),
+			can_edit_generated = VALUES(can_edit_generated),
+			is_owner_generated = VALUES(is_owner_generated)`
+	stmtUpdatePermissionsGenerated, err = s.db.CommonDB().Prepare(queryUpdatePermissionsGenerated)
+	mustNotBeError(err)
+	defer func() { mustNotBeError(stmtUpdatePermissionsGenerated.Close()) }()
+
+	// marking 'self' permissions_propagate as 'children'
+	const queryMarkSelfAsChildren = `
+		UPDATE permissions_propagate
+		SET propagate_access = 'children'
+		WHERE propagate_access = 'self'`
+	stmtMarkSelfAsChildren, err = s.db.CommonDB().Prepare(queryMarkSelfAsChildren)
+	mustNotBeError(err)
+	defer func() { mustNotBeError(stmtMarkSelfAsChildren.Close()) }()
+
+	// ------------------------------------------------------------------------------------
+	// Here we execute the statements
+	// ------------------------------------------------------------------------------------
+	hasChanges := true
+	for hasChanges {
+		_, err = stmtMarkChildrenOfChildrenAsSelf.Exec()
+		mustNotBeError(err)
+		_, err = stmtDeleteProcessedChildren.Exec()
+		mustNotBeError(err)
+		_, err = stmtUpdatePermissionsGenerated.Exec()
+		mustNotBeError(err)
+
+		var result sql.Result
+		result, err = stmtMarkSelfAsChildren.Exec()
+		mustNotBeError(err)
+		var rowsAffected int64
+		rowsAffected, err = result.RowsAffected()
+		mustNotBeError(err)
+		hasChanges = rowsAffected > 0
+	}
```

Comment:
Great, with this proper key ((item_id,group_id) instead the artifical id) and this external table for propagation, we saved a lot of queries and simplified the existing queries.. (and the custom chapter disappearing) 👍